scvi/dataloaders/_contrastive_dataloader.py

```diff
@@ -0,0 +1,127 @@
+import warnings
+from itertools import cycle
+from typing import List, Optional, Union
+
+from scvi import settings
+from scvi.data import AnnDataManager
+
+from ._concat_dataloader import ConcatDataLoader
+
+
+class _ContrastiveIterator:
+    """Iterator for background and target dataloader pair in contrastive analysis.
+
+    Each iteration of this iterator returns a dictionary with two elements:
+    "background", containing one batch of data from the background dataloader, and
+    "target", containing one batch of data from the target dataloader.
+    """
+
+    def __init__(self, background, target):
+        self.background = iter(background)
+        self.target = iter(target)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        bg_samples = next(self.background)
+        tg_samples = next(self.target)
+        return {"background": bg_samples, "target": tg_samples}
+
+
+class ContrastiveDataLoader(ConcatDataLoader):
```

Comment:
If I'm not mistaken, the behavior of this dataloader is identical to `ConcatDataLoader` except that the output is a dictionary with keys background and target. Is this necessary for the logic of the code? If not, it would be preferable to reuse `ConcatDataLoader` directly in `ContrastiveDataSplitter` and modify `ContrastiveVAE._get_inference_input_from_concat_tensors` and `ContrastiveVAE._get_generative_input_from_concat_tensors` to index into the mini-batched output correctly.

Reply:
This is not necessary for the code logic, but in my opinion the dictionary helps with readability and bug prevention in `ContrastiveVAE`. Otherwise it can be easy to swap the integer indices.